### PR TITLE
Added useGravity support

### LINKED ISSUE #75: useGravity is not working
**Describe the bug**
useGravity is not working

**To Reproduce**
1. Set useGravity to true
2. Try to translate the craft
3. Craft will not go down

**Expected behavior**
Craft should fall down, but doesn't

### PR DESCRIPTION
**Pull request by drfiveminusmint**
### Describe in detail what your pull request acomplishes
Added support for the useGravity feature of crafts. Crafts with this feature cannot be controlled vertically by the player, instead, crafts will attempt to "step" up or down sloped terrain when flown into it, and will float towards the ground when moving in the air.

Currently, craft maximum step height is calculated as 50% of the craft's total height, rounded up

### Related issues:
- This will fix #75 

### Checklist
- [X] Proper internationalization
- [X] Compiled/tested

### CODE REVIEW

**Comment by drfiveminusmint** [comment]
@cccm5 could you give this a review?

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java [hunk 1]
```diff
@@ -48,9 +50,27 @@ public TranslationTask(Craft c, int dx, int dy, int dz) {
...
     @Override
-    protected void execute() {
-
-        //Check if theres anything to move
+    protected void execute() {    	
+    	//useGravity handling
+    	if(getCraft().getType().getUseGravity() && checkUseGravity){
+    		failed = true;
+    		suppressMessages = true;
+    		int j = Math.round((oldHitBox.getMaxY()-oldHitBox.getMinY()) / 2.0f);
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java]: use `HitBox::getYLength()`

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java [hunk 2]
```diff
@@ -48,9 +50,27 @@ public TranslationTask(Craft c, int dx, int dy, int dz) {
...
+    		int i = -j;
+    		while(failed && i <= j) { //find a level to "step up" to, or give up if beyond step limit
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java]: use a do while

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java [hunk 3]
```diff
@@ -48,9 +50,27 @@ public TranslationTask(Craft c, int dx, int dy, int dz) {
...
+    			this.failed = false;
+    			dy = i;
+    			if (i == j) { 
+    				suppressMessages = false;
+    			}
+    			this.checkUseGravity = false;
+    			this.execute();
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java]: don't do this

File: modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java [hunk 4]
```diff
@@ -48,9 +50,27 @@ public TranslationTask(Craft c, int dx, int dy, int dz) {
...
+    			newHitBox = new HashHitBox();
```
Comment (cccm5) [inline modules/Movecraft/src/main/java/net/countercraft/movecraft/async/translation/TranslationTask.java]: don't do this

**Comment by drfiveminusmint** [comment]
@cccm5 recursion has been removes and crafts will now descend correctly to the original level. I don't know what's up with Travis, it took ~50 minutes to build because it kept failing to download files but it works fine on my machine.

**Pull request merged**